pkg/scheduler/internal/queue/scheduling_queue_test.go

```diff
@@ -2033,17 +2033,34 @@ var (
 	add = func(logger klog.Logger, queue *PriorityQueue, pInfo *framework.QueuedPodInfo) {
 		queue.Add(logger, pInfo.Pod)
 	}
+	expectNoError = func(what string, err error) {
+		if err != nil {
+			panic(fmt.Sprintf("Unexpected error from %s: %v", what, err))
+		}
+	}
 	addUnschedulablePodBackToUnschedulablePods = func(logger klog.Logger, queue *PriorityQueue, pInfo *framework.QueuedPodInfo) {
 		// To simulate the pod is failed in scheduling in the real world, Pop() the pod from activeQ before AddUnschedulableIfNotPresent() below.
-		queue.activeQ.Add(queue.newQueuedPodInfo(pInfo.Pod))
-		if p, err := queue.Pop(); err != nil || p.Pod != pInfo.Pod {
+		expectNoError("Add", queue.activeQ.Add(queue.newQueuedPodInfo(pInfo.Pod)))
+		p, err := queue.Pop()
+		expectNoError("Pop", err)
+		if p.Pod != pInfo.Pod {
 			panic(fmt.Sprintf("Expected: %v after Pop, but got: %v", pInfo.Pod.Name, p.Pod.Name))
 		}
-
-		queue.AddUnschedulableIfNotPresent(logger, pInfo, 1)
+		expectNoError("AddUnschedulableIfNotPresent", queue.AddUnschedulableIfNotPresent(logger, p, 1))
 	}
 	addUnschedulablePodBackToBackoffQ = func(logger klog.Logger, queue *PriorityQueue, pInfo *framework.QueuedPodInfo) {
-		queue.AddUnschedulableIfNotPresent(logger, pInfo, 1)
+		// To simulate the pod is failed in scheduling in the real world, Pop() the pod from activeQ before AddUnschedulableIfNotPresent() below.
+		expectNoError("Add", queue.activeQ.Add(queue.newQueuedPodInfo(pInfo.Pod)))
+		p, err := queue.Pop()
+		expectNoError("Pop", err)
+		if p.Pod != pInfo.Pod {
+			panic(fmt.Sprintf("Expected: %v after Pop, but got: %v", pInfo.Pod.Name, p.Pod.Name))
+		}
+
+		// A concurrent event forces the pod into the backoff queue instead of the unschedulable queue.
+		queue.MoveAllToActiveOrBackoffQueue(logger, NodeAdd, nil, nil, nil)
```

Comment:
I don't like the fact that, to understand what each test case is doing, I have to go back and read this function.
I would prefer if we would just have:

operations: []{pop, moveAll, addUnschedulable, ...}

But up to you if you want to do it or leave it for someone else to follow up.

Reply:
Someone else, please :sweat: